genie/process_mutation.py

```diff
@@ -226,18 +227,110 @@ def process_mutation_workflow(
         syn=syn,
         center=center,
         maf_tableid=maf_tableid,
-        annotated_maf_path=annotated_maf_path,
+        annotated_maf_path=annotation_paths.merged_maf_path,
         flatfiles_synid=flatfiles_synid,
         workdir=workdir,
     )
 
-    return annotated_maf_path
+    full_error_report = concat_annotation_error_reports(
+        center=center,
+        input_dir=annotation_paths.error_dir,
+    )
+
+    store_annotation_error_reports(
+        full_error_report=full_error_report,
+        syn=syn,
+        errors_folder_synid=genie_config["center_config"][center]["errorsSynId"],
+        release="test",
+    )
+    return annotation_paths.merged_maf_path
+
+
+def create_annotation_paths(center: str, workdir: str) -> namedtuple:
+    """Creates the filepaths required in the annotation process
+
+    Args:
+        center (str): name of the center
+        workdir (str): work directory to create paths in
+
+    Returns:
+        namedtuple: tuple with all the paths
+    """
+    input_files_dir = tempfile.mkdtemp(dir=workdir)
+    output_files_dir = tempfile.mkdtemp(dir=workdir)
+    Filepaths = namedtuple(
+        "Filepaths",
+        ["input_files_dir", "output_files_dir", "error_dir", "merged_maf_path"],
+    )
+    annotation_paths = Filepaths(
+        input_files_dir=input_files_dir,
+        output_files_dir=output_files_dir,
+        error_dir=os.path.join(output_files_dir, f"{center}_error_reports"),
+        merged_maf_path=os.path.join(
+            output_files_dir, f"data_mutations_extended_{center}.txt"
+        ),
+    )
+    return annotation_paths
+
+
+def concat_annotation_error_reports(
+    center: str,
+    input_dir: str,
+) -> pd.DataFrame:
+    """Concatenates the annotation error reports
+
+    Args:
+        center (str): name of center associated with error report
+        input_dir (str): directory where error reports are
+    Returns:
+        pd.DataFrame: full annotation error report
+    """
+    error_files = os.listdir(input_dir)
+    chunk_size = 10000
+    error_reports = []
+
+    # Read and concatenate TSV files in chunks
+    for file in error_files:
+        for chunk in pd.read_csv(
+            os.path.join(input_dir, file), sep="\t", chunksize=chunk_size
+        ):
+            error_reports.append(chunk)
+    full_error_report = pd.concat(error_reports)
+    full_error_report["Center"] = center
+    return full_error_report
+
+
+def store_annotation_error_reports(
+    full_error_report: pd.DataFrame,
+    syn: Synapse,
+    errors_folder_synid: str,
+    release: str,
```

Comment:
I don't see anything related to a processing run version comment as part the parameters available for running `input_to_database`. 

That only occurs with running the consortium release and so forth. I'm guessing what we probably want here for consistency with our other files is for when we run consortium release / public release code to have a function there to comment on the version of the annotation error report (I'm guessing we will always pull the latest version) and mark it as 15.3 consortium release for example? Over trying to add that in during processing here.

Reply:
Hmm you do make a good point.  I think once we get around to separating validation and processing, we will be able to split up the `release` parameter better.  I think it would make this PR overly complex if we were to add in the `release` parameter here.  We do keep track of when the processing code is executed, so the error reports are just updated whenever the processing is run.

To make validation + processing truly idempotent, ideally we are passing in a "manifest" of entities with it's versions as the input.  I did have a question: what happens when there are no annotation errors, is an empty file with headers uploaded?

Reply:
Sounds good to the `release` parameter + future validation + processing thoughts.


> what happens when there are no annotation errors, is an empty file with headers uploaded?

Yes, when there are no annotation errors, it's just an empty file with headers uploaded (gn tool does this and I just decided to keep it consistent with them). Is this preferred behavior? Given that we will have a summary table telling sites if they have any annotation errors at all.

Reply:
That is ok.  No harm in uploading an empty file.